.github/workflows/ci.yml

```diff
@@ -1,50 +1,38 @@
-name: CI
-
+name: build
 on:
-  push:
+  pull_request:
     branches:
       - master
-  pull_request:
+  push:
     branches:
       - master
-
 jobs:
-
   build:
-
     strategy:
-
-      fail-fast: true
-
+      fail-fast: false
       matrix:
-
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
         ocaml-version:
           - 4.14.0
-          - 4.13.1
-          - 4.12.1
-          - 4.11.2
-          - 4.10.2
-          - 4.09.1
-          - 4.08.1
```

Comment:
I agree with checking less versions but I think we want to stay compatible with OCaml 4.08